src/align/hts/bam/reader.jl

```diff
@@ -10,6 +10,19 @@ type BAMReader <: Bio.IO.AbstractReader
     index::Nullable{BAI}
 end
 
+function BAMReader(input::IO; index=nothing)
+    if isa(index, AbstractString)
```

Comment:
`String` is a subtype of `AbstractString` so `isa(index, AbstractString)` returns `true` when `index` is `String`.